src/content/symbol-store.js

```diff
@@ -14,70 +30,86 @@ export class SymbolStore {
-      if (this._importingLibs.has(libid)) {
-        // We've already requested a symbol table for this library and are
-        // waiting for the result, so just return the promise for the existing
-        // request.
-        return this._importingLibs.get(libid);
-      }
 
+    const existingRequest = this._requestedSymbolTables.get(libid);
+    if (existingRequest !== undefined) {
+      // We've already requested a symbol table for this library and are
+      // waiting for the result, so just return the promise for the existing
+      // request.
+      return existingRequest;
+    }
+
+    // Try to get the symbol table from the database
+    const symbolTablePromise = this._db.getSymbolTable(debugName, breakpadId).catch(() => {
       // Request the symbol table from the symbol provider.
-      const symbolTablePromise = this._symbolProvider.requestSymbolTable(lib.debugName, lib.breakpadId);
-      const importPromise = symbolTablePromise.then(symbolTable => {
-        this._importingLibs.delete(libid);
-        return this._db.importLibrary(debugName, breakpadId, symbolTable);
-      }, error => {
-        this._importingLibs.delete(libid);
+      const symbolTablePromise = this._symbolProvider.requestSymbolTable(debugName, breakpadId).catch(error => {
         console.error(`Failed to symbolicate library ${debugName}: ${error}`);
-        throw error;
+        this._failedRequests.add(libid);
+        this._requestedSymbolTables.delete(libid);
+      });
+
+      // Once the symbol table comes in, store it in the database, but don't
+      // let that block the promise that we return to our caller.
+      symbolTablePromise.then(symbolTable => {
+        this._db.storeSymbolTable(debugName, breakpadId, symbolTable).then(() => {
+          this._requestedSymbolTables.delete(libid);
+        }, error => {
+          console.error(`Failed to store the symbol table for ${debugName} ${breakpadId} in the database:`, error);
+        });
       });
-      this._importingLibs.set(libid, importPromise);
-      return importPromise;
-    }).catch(error => {
-      this._failedRequests.add(libid);
-      throw error;
+
+      return symbolTablePromise;
     });
+    this._requestedSymbolTables.set(libid, symbolTablePromise);
+    return symbolTablePromise;
   }
 
   /**
    * Get the array of symbol addresses for the given library.
-   * @param  {object} lib A library object with the properties `debugName` and `breakpadId`.
-   * @return {Promise<Array<Number>>} A promise of the array of addresses.
+   * @param  {Library} lib A library object with the properties `debugName` and `breakpadId`.
+   * @return {Promise<number[]>} A promise of the array of addresses.
    */
-  getFuncAddressTableForLib(lib) {
-    return this._ensureLibraryIsInDB(lib).then(
-      libKey => this._db.getFuncAddressTableForLib(libKey));
+  getFuncAddressTableForLib(lib: Library): Promise<Uint32Array> {
+    return this._getSymbolTable(lib).then(([addrs]) => addrs);
```

Comment:
are you sure this is `[addrs]` and not `{addrs}` ? Just basing my logic on what I see in the old code in https://github.com/devtools-html/perf.html/pull/276/files#diff-e1ccc283046ee435c73e933d577ebcbaL85 so I may be wrong, but double-checking :)